### ISSUE TITLE
Multilingual support in "New post" RStudio addin

### ISSUE THREAD

**Issue by Guilz** [rstudio/blogdown#323]
One nice thing about Hugo is support for multilingual sites. With a theme making use of this feature, one can specify the language of a post very simply in the file name: `2018-08-10-my-new-post.md` will be assigned the default language by Hugo, while `2018-08-10-my-new-post.<langcode>.md` will be assigned the correct language corresponding to the 2-letter `<langcode>`, and Hugo will place the corresponding html page in `<baseURL>/<langcode>/...` when it builds the site.

Now, when writing a new post I almost always forget to specify the language code in the file name, notice it when previewing the site locally, and fix it. I feel like I would not forget it as easily if the "New post" addin in RStudio had a dropdown menu with available languages (and would construct the filename accordingly for me, just like it already does with the date).
A reasonable default for this menu would be the value of `defaultContentLanguage` in `config.toml`, and the menu could auto-detect other language codes in use in already existing files (just like existing tags and categories are detected by the addin). Or alternatively, make it "smart" and default to the most commonly used language code based on already existing posts? Could be nice too.

Any chance this little feature could be added to the "New post" addin? If so, that would be great. 😃

**Comment by yihui** [comment]
I agree that will be very useful, but I don't have time to implement this by myself. I can probably review a pull request if you or someone else (@lcolladotor?) can help implement it. Thanks!

**Comment by sumendar** [comment]
haiiii

**Comment by Guilz** [comment]
Won't be me I'm afraid: I have zero experience with Shiny, and after taking a look at `inst/scripts/new_post.R` I don't think I can figure it out in the little free time I have these days.

**Comment by novica** [comment]
Hi, 

I made some progress on this issue. The diff is [here](https://github.com/rstudio/blogdown/compare/master...novica:multilingual-new-post).
I'm still trying to figure out how to make the addin read the .toml file for default values. Other than that the filename formatting is not perfect. 
I tested the changes on my computer and they don't break the package.

**Comment by yihui** [comment]
@novica Sounds great. Could you post a screenshot of the revised addin? Thanks!

**Comment by novica** [comment]
![screenshot_20181118_105455](https://user-images.githubusercontent.com/165724/48670901-70474580-eb20-11e8-85d5-84eaf3f84920.png)

Here it is.

**Comment by yihui** [comment]
Thanks! I'd like to get rid of the vertical scrollbar. Could you move `Language` to the right side of `Slug`?

**Comment by novica** [comment]
@yihui I moved the field as you suggested. I also added code to check if there is actually multilingual options in the `toml` file so that the field shows only if the user has set up `DefaultContentLanguage`. If not the the `Language` field does not show.
I think it works as it is supposed to.

**Comment by yihui** [comment]
Perfect. Please submit a PR when you have a chance. Thank you!

**Comment by MartinsRodrigo** [comment]
This solution is already implemented?

**Comment by Guillawme** [comment]
> This solution is already implemented?

Yes, it was realeased in [v0.10](https://github.com/rstudio/blogdown/releases/tag/v0.10).

Now, it seems that the following line:
https://github.com/rstudio/blogdown/blob/71881a7a8af996051ea5dc94b0efbae80cf0c96e/R/utils.R#L158
will look for a definition of `DefaultContentLanguage` in your `config.toml` file in a case-sensitive manner. This means that if your `config.toml` contains `defaultContentLanguage` (no initial capital), the add-on will not detect that your site is multi-lingual and will not display the text field to change the language code.

It seems Hugo doesn't care about the capitalization of parameters in `config.toml`, so the fix is easy: capitalize your `DefaultContentLanguage` parameter and you will see the text field for the language code (I just tried, I see this text field).
